src/modules/types.rs

```diff
@@ -64,15 +82,19 @@ pub fn try_parse_type(meta: &mut ParserMetadata) -> Result<Type, Failure> {
                 "[" => {
                     let index = meta.get_index();
                     meta.increment_index();
-                    match try_parse_type(meta) {
-                        Ok(Type::Array(_)) => error!(meta, tok, "Arrays cannot be nested due to the Bash limitations"),
-                        Ok(result_type) => {
-                            token(meta, "]")?;
-                            Ok(Type::Array(Box::new(result_type)))
-                        },
-                        Err(_) => {
-                            meta.set_index(index);
-                            Err(Failure::Quiet(PositionInfo::at_eof(meta)))
+                    if token(meta, "]").is_ok() {
+                        Ok(Type::Array(Box::new(Type::Generic)))
+                    } else {
+                        match try_parse_type(meta) {
+                            Ok(Type::Array(_)) => error!(meta, tok, "Arrays cannot be nested due to the Bash limitations"),
+                            Ok(result_type) => {
+                                token(meta, "]")?;
+                                Ok(Type::Array(Box::new(result_type)))
+                            },
+                            Err(_) => {
+                                meta.set_index(index);
+                                Err(Failure::Quiet(PositionInfo::at_eof(meta)))
+                            }
                         }
```

Comment:
Try to limit indents. Code will be easier to follow

```suggestion
                    if token(meta, "]").is_ok() {
                        return Ok(Type::Array(Box::new(Type::Generic)));
                    }
                    match try_parse_type(meta) {
                        Ok(Type::Array(_)) => error!(meta, tok, "Arrays cannot be nested due to the Bash limitations"),
                        Ok(result_type) => {
                            token(meta, "]")?;
                            Ok(Type::Array(Box::new(result_type)))
                        },
                        Err(_) => {
                            meta.set_index(index);
                            Err(Failure::Quiet(PositionInfo::at_eof(meta)))
                        }
                        
```

Reply:
We cannot early return from that block, the function has additional logic at the end. I cannot easily refactor it either.

Reply:
But these are identical when it comes to logic. You already return Ok erly. I just want you to not do else block, because it's unnecessary

Reply:
I do not return Ok early, please pay attention to the code. The if-else block is in a match arm, and the match expression is not the last thing in the function. If I return early, the failable logic located after the match expression won't run.